packages/flutter_test/lib/src/controller.dart

```diff
@@ -761,6 +761,12 @@ abstract class WidgetController {
       kind: kind,
       buttons: buttons,
     );
+
+    final size = binding.window.physicalSize / binding.window.devicePixelRatio;
+    if (!size.contains(downLocation)) {
+      print('Trying to tap non-visible area.');
```

Comment:
I'm pretty sure this is incorrect - `tap` is expecting logical, not physical pixels. 

This print is also likely to be missed/ignored by developers or CI.

I think the idea here could be valuable, but we'd probably want it to be opt-in (or perhaps opt-out), so the developer could choose to have a `fail` here if the tap is outside the tappable area of the screen.